src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file SignalGeneratorComponent.cpp
+ * @author Luciana Moreno Borges (lumoreno.borges@gmail.com)
+ * @brief Signal Generator component
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/msgs/common/Basic.h>
+#include <../src/internal/SignalGenerator.h>
+#include <hyro/msgs/Signal.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <iostream>
+
+namespace hyro
+{
+  /* Create a log variable */
+  std::shared_ptr<hyro::HyroLogger> SignalGeneratorComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+  
+  /*Initialize the object output and register the dynamic properties*/
+  hyro::Result
+  SignalGeneratorComponent::init (const hyro::ComponentConfiguration & config)
+  {
+    m_signal = SignalGenerator();
+    std::shared_ptr<ChannelOutput<std::vector<int>>>
+    m_dummy = this->registerOutput<std::vector<int>>("fix_dynamic"_uri, config);
+    s_logger->info("Init:");
+  
+    m_output = this->registerOutput<Signal>("signals"_uri, config);
+
+    m_amplitude = 5.0f;
```

Comment:
Get the parameters from the configuration file, with default values, just like you did in `DigitalConverterComponent.cpp`